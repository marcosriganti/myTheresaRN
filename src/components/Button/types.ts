import {PressableProps} from "react-native";
type ButtonType = 'primary' | 'secondary' | 'tertiary';
interface ButtonProps extends PressableProps {
    type?: ButtonType;
}

export type {
    ButtonProps,
    ButtonType,
};
