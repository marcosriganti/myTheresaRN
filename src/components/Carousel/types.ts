export type Record = {
    name?: string;
    backdrop_path: string;
    id: number;
    title: string;
    original_title: string;
    overview: string;
    poster_path: string;
    media_type: string;
    adult: boolean;
    popularity: number;
    release_date: string;
    video: boolean;
    original_language: string;
    genre_ids: number[];
    vote_average: number;
    vote_count: number;
};

export interface CarouselProps {
    title: string;
    data: Record[];
}
