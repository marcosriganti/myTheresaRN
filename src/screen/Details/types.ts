import type {NativeStackScreenProps} from '@react-navigation/native-stack';
import {RootStackParamList} from '../../types';

type DetailsScreenProps = NativeStackScreenProps<RootStackParamList, 'Details'>;
export type {
    DetailsScreenProps,
};
