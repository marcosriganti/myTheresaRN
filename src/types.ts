export interface Category {
    id: number;
    name: string;
}
export type Record = {
    name?: string;
    backdrop_path: string;
    id: number;
    title: string;
    original_title: string;
    overview: string;
    poster_path: string;
    media_type: string;
    adult: boolean;
    popularity: number;
    release_date: string;
    video: boolean;
    original_language: string;
    genre_ids: number[];
    vote_average: number;
    vote_count: number;
};
export interface queryResult {
    'page': number,
    'results': Record[],
    'total_pages': number,
    'total_results': number;
}

export type RootStackParamList = {
    Home: undefined;
    Details: Record;
};
