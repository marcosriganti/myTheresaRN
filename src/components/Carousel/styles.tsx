import {
    Text,
    Pressable,
    Image,
    View,
} from 'react-native';
import styled from 'styled-components';

const RADIUS = '16px';

const ItemTitleWrapper = styled(View) <{borderColor?: string;}>`
    background-color: rgba(0,0,0,0.5);
    border-bottom-color: ${props => props.borderColor ? props.borderColor : '#0B1E32'};
    border-bottom-width: 3px;
    position: absolute;
    left: 0;
    z-index: 2;
    width: 100%;
    text-align: center;
`;
const CarouselTitle = styled(Text)`
    font-size: 16px;
    color: white;
    font-weight: bold;
    font-style: italic;
    padding-left: 8px;
    padding-right: 8px;
`;
const ItemTitle = styled(CarouselTitle)`
    font-size: 16px;
    color: #fff;
    padding-top: 8px;
    padding-bottom: 8px;
    overflow: hidden;
    text-align: center;
`;
const EmptyDataText = styled(Text)`
    font-size: 12px;
    font-weight: bold;
    font-style: italic;
    color: white;

`;
const EmptyData = styled(View)`
    padding: 16px;
    background-color: rgba(0,0,0,0.5);
    margin:  16px;
    border-radius: ${RADIUS};
    
`;
const StyledShadow = styled(View) <{isDark?: boolean;}>`
    border-radius: ${RADIUS};
    margin: 10px;
    background-color: rgba(0,0,0,0.5);
    background-color: #0B1E32;
    shadow-color: ${props => !props.isDark ? '#000' : '#CCC'};
    shadow-offset: {
        width: 0;
        height: 3,
    };
    shadow-opacity: 0.27;
    shadow-radius: 4.65px;
    elevation: 4;
`;
const CarouselItem = styled(Pressable)`
    border-style: "solid";
    border-width: 1px;
    width: 200px; 
    height: 350px;
    overflow: hidden;
    position: relative;
    border-radius: ${RADIUS};
`;
const StyledImage = styled(Image)`
    border-radius: 8px; 
    resize-mode: cover;
    width: 200px;
    height: 350px;
`;
export {
    ItemTitle,
    CarouselTitle,
    CarouselItem,
    StyledShadow,
    StyledImage,
    EmptyData,
    ItemTitleWrapper,
    EmptyDataText,
};
