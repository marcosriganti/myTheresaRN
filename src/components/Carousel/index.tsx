import React from 'react';
import {
    ScrollView,
    useColorScheme,
    View,
} from 'react-native';
import {CarouselItem, CarouselTitle, ItemTitle, StyledShadow, StyledImage, EmptyData, ItemTitleWrapper, EmptyDataText} from './styles';

import {Record, Category} from '../../types';

interface CarouselProps {
    title: string;
    data: Record[];
    category?: Category;
    onPress: (item: Record, category: Category) => void;
}

const Carousel = ({title, data, category, onPress}: CarouselProps) => {
    const isDarkMode = useColorScheme() === 'dark';
    return <View>
        <CarouselTitle>{title}</CarouselTitle>
        <View>
            {data.length === 0 && <EmptyData><EmptyDataText>No films to display in here</EmptyDataText></EmptyData>}
            <ScrollView horizontal={true}>
                {data.map((item) => {
                    return <StyledShadow isDark={isDarkMode} key={`item-${item.id}`}>
                        <CarouselItem key={item.id} onPress={() => category && onPress(item, category)}>
                            <ItemTitleWrapper borderColor={category?.backgroundColor}><ItemTitle>{item.name || item.original_title}</ItemTitle></ItemTitleWrapper>
                            <StyledImage source={{
                                uri: `https://image.tmdb.org/t/p/w300${item.poster_path}`,
                            }} />
                        </CarouselItem>
                    </StyledShadow>;
                })}
            </ScrollView>
        </View>
    </View >;
};
export const Skeleton = (props: {title: string;}) => {
    return <View>
        <CarouselTitle>{props.title}</CarouselTitle>
        <View>
            <ScrollView horizontal={true}>
                {[1, 2, 3, 4, 5].map((i) => {
                    return <StyledShadow key={`item${i}`} isDark={false}>
                        <CarouselItem>
                            <ItemTitle>loading...</ItemTitle>
                            {/* something */}
                        </CarouselItem>
                    </StyledShadow>;
                })}
            </ScrollView>
        </View>
    </View>;
};
export default Carousel;
