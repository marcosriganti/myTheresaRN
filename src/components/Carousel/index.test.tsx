import 'react-native';
import React from 'react';
import Carousel from './index';
import {render, screen} from '@testing-library/react-native';
import {Record, Category} from '../../types';

describe('Carousel', () => {
    const mockData: Record[] = [
        {id: '1', name: 'Movie 1', original_title: 'Original Movie 1', poster_path: '/path1.jpg'},
        {id: '2', name: 'Movie 2', original_title: 'Original Movie 2', poster_path: '/path2.jpg'},
    ];

    const mockCategory: Category = {
        id: 1,
        name: 'Category 1',
        backgroundColor: '#FF0000',
    };

    const mockOnPress = jest.fn();

    it('renders correctly', () => {
        render(<Carousel title="Test Carousel" data={mockData} category={mockCategory} onPress={mockOnPress} />);
        expect(screen.toJSON()).toMatchSnapshot();
    });

    it('displays the title', () => {
        render(<Carousel title="Test Carousel" data={mockData} category={mockCategory} onPress={mockOnPress} />);
        expect(screen.getByText('Test Carousel')).toBeTruthy();
    });

    it('displays empty data text when data is empty', () => {
        render(<Carousel title="Test Carousel" data={[]} category={mockCategory} onPress={mockOnPress} />);
        expect(screen.getByText('No films to display in here')).toBeTruthy();
    });

    it('displays items when data is provided', () => {
        render(<Carousel title="Test Carousel" data={mockData} category={mockCategory} onPress={mockOnPress} />);
        expect(screen.getByText('Movie 1')).toBeTruthy();
        expect(screen.getByText('Movie 2')).toBeTruthy();
    });
});
