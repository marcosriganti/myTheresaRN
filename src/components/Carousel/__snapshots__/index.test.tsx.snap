// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel renders correctly 1`] = `
<View>
  <Text
    className="sc-gtLWhw extFkf"
  >
    Test Carousel
  </Text>
  <View>
    <RCTScrollView
      horizontal={true}
    >
      <View>
        <View
          className="sc-ivxoEo iMwyCZ"
          isDark={false}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            className="sc-ghWlax edfImK"
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <View
              borderColor="#FF0000"
              className="sc-blHHSb Szlbs"
            >
              <Text
                className="sc-gtLWhw sc-egkSDF extFkf bIgjLZ"
              >
                Movie 1
              </Text>
            </View>
            <Image
              className="sc-kLhKbu ZsPEz"
              source={
                {
                  "uri": "https://image.tmdb.org/t/p/w300/path1.jpg",
                }
              }
            />
          </View>
        </View>
        <View
          className="sc-ivxoEo iMwyCZ"
          isDark={false}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            className="sc-ghWlax edfImK"
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <View
              borderColor="#FF0000"
              className="sc-blHHSb Szlbs"
            >
              <Text
                className="sc-gtLWhw sc-egkSDF extFkf bIgjLZ"
              >
                Movie 2
              </Text>
            </View>
            <Image
              className="sc-kLhKbu ZsPEz"
              source={
                {
                  "uri": "https://image.tmdb.org/t/p/w300/path2.jpg",
                }
              }
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
